### PR TITLE
Fix #1446 - Configure the grpc server for the external authz

### LINKED ISSUE #1446: [Envoy-MG] Implement filter chain gRPC server capabilities and filter handler architecture
### Describe your problem(s)
The filter chain server should be implemented in order to handle concurrent requests. And also pluggable filter chain architecture should be implemented

### Describe your solution
The gRPC server can be implemented using the netty gRPC support and the interface should be introduced in order to support implement the handler.filter chain capabilities.

### How will you implement it

---

### PR DESCRIPTION
**Pull request by Rajith90**
### Purpose
This adds the configurations for the gRPC netty server that serves the external authz requests coming from the envoy side.
### Issues

Fixes #1446 

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: enforcer/src/main/java/org/wso2/micro/gateway/enforcer/config/EnforcerConfig.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/*
```
Comment (VirajSalaka) [inline enforcer/src/main/java/org/wso2/micro/gateway/enforcer/config/EnforcerConfig.java:1]: License header missing.

File: enforcer/src/main/java/org/wso2/micro/gateway/enforcer/config/ConfigHolder.java [hunk 2]
```diff
@@ -146,19 +129,17 @@ private void loadTrustStore() {
...
-    public EventHubConfigurationDto getEventHubConfiguration() {
-        return eventHubConfiguration;
+    public void setConfig(EnforcerConfig config) {
+        this.config = config;
     }
 
     public KeyStore getTrustStore() {
         return trustStore;
     }
+
```
Comment (VirajSalaka) [inline enforcer/src/main/java/org/wso2/micro/gateway/enforcer/config/ConfigHolder.java:143]: extraneous lines

File: enforcer/src/main/java/org/wso2/micro/gateway/enforcer/config/dto/AuthServiceConfigurationDto.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+        public int getQueueSize() {
+            return queueSize;
+        }
+
+        public void setQueueSize(int queueSize) {
+            this.queueSize = queueSize;
+        }
+    }
+
```
Comment (VirajSalaka) [inline enforcer/src/main/java/org/wso2/micro/gateway/enforcer/config/dto/AuthServiceConfigurationDto.java:113]: extraneous line

File: resources/conf/config.toml [hunk 4]
```diff
@@ -42,13 +55,16 @@ password = "wso2carbon"
...
+    [enforcer.eventHub.jmsConnectionParameters]
+        # The message broker connection URL.
+        eventListeningEndpoints = "amqp://admin:admin@carbon/carbon?brokerlist='tcp://localhost:5672'"
+
+[enforcer.apimCredentials]
     username="admin"
-    password="admin"
-    # The message broker connection URL.
-    eventListeningEndpoints = "amqp://admin:admin@carbon/carbon?brokerlist='tcp://localhost:5672'"
+    password=['a','d','m','i','n']
```
Comment (VirajSalaka) [inline resources/conf/config.toml:70]: for the password, shall can't we use a string ?
new line missing

Comment (AmaliMatharaarachchi) [reply resources/conf/config.toml:70]: I too think we should keep this as a string and convert it to charset at TOML parser level.

**Comment by Rajith90** [comment]
Sent another PR: https://github.com/wso2/product-microgateway/pull/1537

**Pull request closed without merging**